src/idpconfgen/cli_build.py

```diff
@@ -1222,6 +1226,12 @@ def conformer_generator(
                 # add the carboxyls
                 template_coords[TEMPLATE_MASKS.cterm] = \
                     MAKE_COORD_Q_COO_LOCAL(bb[-2, :], bb[-1, :])
+                    
+            except ValueError:
+                # Int2Cart requires length of seq to match length of torsions
+                # if there is a mismatch match is found, start from scratch
```

Comment:
Can you add here just a quick sentence when (in which situations) such a mismatch can happen?

Reply:
Actually the curious thing here is that this error is not consistent, e.g. can still make some conformers but on my machine at around conformer # 30-40 and 60-70, a mismatch occurs... I will do my best to find a consistency though

Reply:
I see one place where this might go wrong: when the backbone energy exceeds threshold, IDPCG algorithm does one backtrack step, and correspondingly the torsion angles for this chunk is removed (see https://github.com/julie-forman-kay-lab/IDPConformerGenerator/blob/main/src/idpconfgen/cli_build.py#L1347). Maybe here it should be `torsion_records[:current_res_number]`?

Reply:
Could you print current_res_num when every time torsion_records changed?

Reply:
Are the list 5 res numbers values for current_res_num in 5 previous iterations? As I understand it current_res_num+1 should be the same as the length of the torsion records, but it does not seem to be the case for what you provided

Reply:
Perhaps it was a bit misleading, the "0" in the "last 5 res numbers" is the current res number. The "last_X" list updates at the beginning of the for-loop before torsion records or current res number changes. If an error occurs, it printed the last 5 items in the "last_X" list.

Reply:
Then I still cannot understand why it can be 5,6,7,8,0. This means starting from a fragment of 5 residues, in the next several iterations only chunks of a single residues are added until it goes up to 8 residues. Then somehow it goes back to zero residue? That does not seem to be the logic of the code

Reply:
Great discussion.
The machinery inside the build should flow without the need for many exceptions. Things around the machinery should adapt to that and provide whatever the machinery needs to continue flowing. Reading from the discussion, I think the bug may come from the `get_adjacent_angles` function. For some reason, that function is giving one set of torsions more than requested. It's a tricky function. Does it happen only with `int2cart`? Or does it happen with other options as well simply the builder ignores that extra set?